### ISSUE TITLE
expressionProperties function returning object does not update in Formly 6

### ISSUE THREAD

**Issue by mikejpeters** [ngx-formly/ngx-formly#3068]
**Description**

In Formly 6 when setting a template option to an object using an `expressionProperties` function,
the value does not update in the field.

e.g. when accessing `to.json` from the field component it will always be the initial value "none", even when `model.input1` has a value:
```ts
  expressionProperties: {
    'templateOptions.json': (model) => {
      return { foo: model.input1 ?? 'none' };
    },
  },
```

This works fine when returning a string or number instead of an object.
Is returning an object supported (it seems to work in Formly 5)?

**Minimal Reproduction**

Here is an example of it working the way I'd expect in Formly 5:
https://stackblitz.com/edit/angular-formly-5-expressions?file=src/app/app.component.ts

This same example should fail when run on Formly 6.0.0-next.1 and 6.0.0-next.4
(I tried to create a stackblitz using Angular 13 and Formly 6 but kept getting errors, is there a starter I can use?)

**Your Environment**

- Angular version: 13.0.2
- Formly version: 6.0.0-next.4

**Additional context**

I tested this on 6.0.0-next.1 and 6.0.0-next.4 and in both cases the value doesn't update.

When I log the template option from the field component using `ngDoCheck()` it looks like this:

```
{}
  foo: (...)
  get foo: () => state.value
  set foo: currentValue => {…}
  [[Prototype]]: Object
```

and if I click on it the getter always evaluates to the initial value I set (e.g. "test")

**Comment by aitboudad** [comment]
> (I tried to create a stackblitz using Angular 13 and Formly 6 but kept getting errors, is there a starter I can use?)

its ok as `stackblitz` doesn't support  [Angular 13](https://github.com/stackblitz/core/issues/1657) yet

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.0.0-next.5](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.0.0-next.5) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!

**Comment by markwhitfeld** [comment]
FYI, StackBlitz now supports Angular 13:
> Great news! Support for Angular 13 has finally rolled out.
...
For the next day or two (until 12 December 2021), there is a possibility of an initial slow down in starting up the dev server or running NGCC (or potentially even timeouts) as the packages repopulate into the cache as they are used, but it should come right after a refresh or two if you do have any issues. Anything else should be reported as a new issue.

Related StackBlitz issue comment: https://github.com/stackblitz/core/issues/1657#issuecomment-989779722
